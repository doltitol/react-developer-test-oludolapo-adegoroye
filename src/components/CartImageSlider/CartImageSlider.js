import React, { PureComponent } from 'react';
import { Icons } from '../../assets/images/Icons';
import './cartimageslider.style.scss';

export class CartImageSlider extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      imageCount: 0,
    };
  }
  handleLeftClick = (count) => {
    this.setState({
      imageCount: count === 0 ? this.props.gallery.length - 1 : count - 1,
    });
  };
  handleRightClick = (count) => {
    this.setState({
      imageCount: count === this.props.gallery.length - 1 ? 0 : count + 1,
    });
  };
  render() {
    return (
      <div className='cart-image-slider' data-testid='cart-image-slider'>
        <img
          src={this.props.gallery[this.state.imageCount]}
          alt=''
          className='cart-image-slider-image'
        />
        {this.props.gallery.length > 1 && (
          <div className='cart-image-slider-buttons'>
            <Icons.LeftArrow
              size={24}
              color='#000000'
              onClick={() => this.handleLeftClick(this.state.imageCount)}
            />
            <Icons.RightArrow
              size={24}
              color='#000000'
              onClick={() => this.handleRightClick(this.state.imageCount)}
            />
          </div>
        )}
      </div>
    );
  }
}

export default CartImageSlider;
