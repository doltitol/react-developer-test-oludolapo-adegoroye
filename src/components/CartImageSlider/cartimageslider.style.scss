.cart-image-slider {
  width: 200px;
  height: 288px;
  position: relative;
  outline: none;
  &-image {
    width: 200px;
    height: 288px;
    object-fit: contain;
    outline: none;
    -webkit-user-select: none;
 -khtml-user-select: none;
 -moz-user-select: none;
 -o-user-select: none;
 user-select: none;
  }
  &-buttons {
    position: absolute;
    bottom: 8px;
    width: 56px;
    right: 8px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;
    outline: none;
  }
}
