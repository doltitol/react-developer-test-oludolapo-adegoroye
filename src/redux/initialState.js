export const initialState = {
  cartItems: JSON.parse(localStorage.getItem('cartItems') || '[]'),
  cartQuantity: JSON.parse(localStorage.getItem('cartItems') || '[]').reduce(
    (acc, item) => acc + item.qty,
    0
  ),
  currency: '$',
};
