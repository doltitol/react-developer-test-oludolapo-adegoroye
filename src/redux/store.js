import { createStore, combineReducers } from 'redux';
import cartReducer from './reducers/cartReducers';

const rootReducer = combineReducers({
  cart: cartReducer,
});
export const store = createStore(rootReducer);
