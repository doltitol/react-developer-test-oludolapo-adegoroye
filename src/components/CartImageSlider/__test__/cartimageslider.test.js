import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import CartImageSlider from '../CartImageSlider';

describe('<CartImageSlider /> spec', () => {
  it('renders the empty component', () => {
    const props = {
      gallery: [
        'https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-12-pro-family-hero?wid=940&amp;hei=1112&amp;fmt=jpeg&amp;qlt=80&amp;.v=1604021663000',
      ],
    };
    const container = render(
      <Provider store={store}>
        <BrowserRouter>
          <CartImageSlider {...props} />
        </BrowserRouter>
      </Provider>
    );
    const getByTestId = container.getByTestId;
    const cartImageSlider = getByTestId('cart-image-slider');
    expect(cartImageSlider.children).toHaveLength(1);
    expect(container.children).toMatchSnapshot();
  });
});
