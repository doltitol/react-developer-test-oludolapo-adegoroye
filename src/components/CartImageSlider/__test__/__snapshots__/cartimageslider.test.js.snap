// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartImageSlider /> spec renders the empty component 1`] = `undefined`;

exports[`<CartOverlayItem /> spec renders the empty component 1`] = `undefined`;
